tpcx_bb/queries/q27/tpcx_bb_query_hf_27.py

```diff
@@ -0,0 +1,161 @@
+### The inference part of workflow to get the token labels are in  q27_bert_utils.run_inference_on_df
+### The sentences are gathered using EOL char as `.`
+### The details for sentence gathering are at q27_get_review_sentence_utils.get_review_sentence
+
+# Current limitation
+### We don't do model based sentence boundary disambiguation
+### We get empty sentence in 0.04% of the cases because of it
+
+import rmm
+import numpy as np
+import os
+import logging
+import time
+
+from xbb_tools.utils import (
+    benchmark,
+    tpcxbb_argparser,
+    run_query,
+)
+
+from xbb_tools.readers import build_reader
+from dask.distributed import Client, wait, get_worker
+
+### Query Specific Utils
+from xbb_tools.q27_bert_utils import (
+    run_inference_on_df,
+    load_model,
+    create_vocab_table,
+    del_model_attribute,
+)
+from xbb_tools.q27_get_review_sentence_utils import get_review_sentence
+
+# -------- Q27 -----------
+q27_pr_item_sk = 10002
+
+
+def read_tables(config):
+    ### splitting by row groups for better parallelism
+    table_reader = build_reader(
+        data_format=config["file_format"],
+        basepath=config["data_dir"],
+        split_row_groups=True,
+    )
+    product_reviews_cols = ["pr_item_sk", "pr_review_content", "pr_review_sk"]
+    product_reviews_df = table_reader.read(
+        "product_reviews", relevant_cols=product_reviews_cols
+    )
+    return product_reviews_df
+
+
+def run_single_part_workflow(df, model_path):
+    """
+    This function runs the entire ner workflow end2end on a single GPU
+    """
+    import cudf
+
+    w_st = time.time()
+
+    worker = get_worker()
+    if hasattr(worker, "q27_model"):
+        model = worker.q27_model
+    else:
+        model = load_model(model_path)
+        worker.q27_model = model
+
+    id2vocab, vocab2id = create_vocab_table(os.path.join(model_path, "vocab.txt"))
+    vocab_hash_file = os.path.join(model_path, "vocab-hash.txt")
+
+    token_d, prediction_d = run_inference_on_df(
+        df, model, vocab_hash_file, batchsize=128
+    )
+
+    output_d = {}
+
+    for seq, pred_label in prediction_d.items():
+        if pred_label is not None:
+            sen_df = get_review_sentence(
+                token_d[seq], prediction_d[seq], vocab2id, id2vocab
+            )
+            review_df = token_d[seq]["df"][["pr_review_sk", "pr_item_sk"]]
+            review_df = review_df.reset_index(drop=False)
+            review_df.rename(columns={"index": "input_text_index"}, inplace=True)
+            output_d[seq] = sen_df.merge(review_df)[
+                ["pr_review_sk", "pr_item_sk", "company_name", "review_sentence"]
+            ]
+
+    del token_d, prediction_d
+
+    output_df = cudf.concat([o_df for o_df in output_d.values()])
+    output_df.rename(
+        columns={"pr_review_sk": "review_sk", "pr_item_sk": "item_sk"}, inplace=True
+    )
+
+    w_et = time.time()
```

Comment:
No problem keeping these, but do we need them?

Reply:
Helps to triage bottlenecks and dask overhead will remove in a followup pr.

Reply:
👍